content/blog/rest-api-monitoring-with-custom-java-flight-recorder-events.asciidoc

```diff
@@ -55,79 +57,71 @@ In sub-classes you can add application-specific payload attributes,
 as well as some metadata such as a name and category which will be used for organizing and displaying events when looking at them in JMC.
 
 Which attributes to add depends on your specific requirements;
-you should aim at finding the right balance between capturing all the relevant information that will be useful for analysis purposes later on, while not going overboard and adding too much,
-as it'd cause record files to become too large, in particular for events that are emitted with a high frequency.
+you should aim for the right balance between capturing all the relevant information that will be useful for analysis purposes later on, while not going overboard and adding too much,
+as that could cause record files to become too large, in particular for events that are emitted with a high frequency.
```

Comment:
I would add here that a good rule is to not consider JFR events as a form of "logging"

Reply:
That's an interesting point, as in fact I was/am kinda seeing it like that. I actually make a related point towards the end:

> Combined with the Event Streaming API introduced in Java 14, this opens up an attractive, very performant alternative to other means of capturing analysis information at application runtime, such as logging libraries.

I'm interested in why you say what you said. I'd agree it'd not be a general form of logging, but it definitely would cover some cases where one otherwise would have used logging. I'm not sure though where to draw the line.

Reply:
Oh, yeah, and you example clearly show one area where they overlap. My point is that even with a very optimised framework like this you still incur in some overhead, and depending on how you use the events this overhead may be more or less pronounced. Most of the defaults events benefit from well thought compromises so at to cost as little as possible, but with custom events the programmer is responsible to keep this balance. For this reason considering this a replacement for java.util.logging is probably a fast track to problems. So my statement above was not very correct, what I meant was specifically to avoid abusing JFR as a replacement for JUL.

Reply:
Ok, gotcha. I think it's ok then as is, as I'm not saying it should be considered as such replacement.